parlai/core/worlds.py

```diff
@@ -550,6 +550,13 @@ def __next__(self):
 
     def batch_observe(self, index, batch_actions, index_acting):
         batch_observations = []
+        a = self.world.get_agents()[index]
+
+        if (batch_actions is not None and len(batch_actions) > 0 and
+                hasattr(a, 'batch_observe')) and index != index_acting:
+            # if there's anything that needs to be batched, do it here
+            a.batch_observe(batch_actions)
```

Comment:
@alexholdenmiller Not exactly. The instance that `batch_observe` is being called on is different from the ones that `observe` would have been called on, which relies on their internal states to update the metrics (`self.lastY`). If `batch_observe` is mutually exclusive with `observe` for agents, all `Teacher`s would have to implement `batch_act` as well and update some batch-shared states for metric update to work?